include/xtensor/xbuffer_adaptor.hpp

```diff
@@ -1188,13 +1255,25 @@ namespace xt
     }
 
     template <class C, class IG>
-    inline auto xiterator_owner_adaptor<C, IG>:: data() noexcept -> iterator
+    inline auto xiterator_owner_adaptor<C, IG>::data() noexcept -> pointer
+    {
+        return &(*m_it);
```

Comment:
Sorry, maybe I need coffee, but is this not either useless, or illegal. Useless because it will return the same memory address. Or, illegal, because it might point to a temporary?

Also: Will this break things for users?

Reply:
This is indeed equivalent to `m_it` when dereferencing `m_it` returns an lvalue, and illgal when it return an rvalue, so this should be reverted to the previous implementation.

Reply:
this was done to make data() consistently return `pointer` instead of `iterator` in this case, but it's indeed faulty and not needed here.